intro-to-R-tidyverse/02-intro_to_ggplot2.Rmd

```diff
@@ -0,0 +1,307 @@
+First let's take a look at only the tumor vs. normal comparison. 
+Let's save this as a separate data.frame by assigning it a new name. 
+
+```{r}
+tumor_normal_df <- filter(stats_df, contrast == "astrocytoma_normal")
+```
+
+To make this plot we will be using functions from the `ggplot2` package, the main plotting package of the tidyverse.
+We use the first function, `ggplot()` to define the data that will be plotted.
+`ggplot()` takes two main arguments:  
+
+1. `data`, which is the data frame that contains the data we want to plot.  
+2. `mapping`, which is a special list made with the `aes()` function to describe which values will be used for each **aes**thetic component of the plot, such as the x and y coordinates of each point. 
+(If you find calling things like the x and y coordinates "aesthetics" confusing, don't worry, you are not alone.)  
+
+```{r}
+ggplot(
+  tumor_normal_df, # This first argument is the data.frame with the data we want to plot
+  aes(
+    x = log_fold_change, # This is the column name of the values we want to use
+    # for the x coordinates
+    y = neg_log_p_val
+  ) # This is the column name of the data we want for the y-axis
+)
+```
+
+You'll notice this plot doesn't have anything on it because we haven't 
+specified a plot type yet. 
+To do that, we will add another ggplot layer with `+` which will specify exactly what we want to plot.
+A volcano plot is a special kind of scatter plot, so to make that we will want to plot individual points, which we can do with `geom_point()`.
+
+```{r}
+# This first part is the same as before
+ggplot(
+  tumor_normal_df,
+  aes(
+    x = log_fold_change,
+    y = neg_log_p_val
+  )
+) +
+  # Now we are adding on a layer to specify what kind of plot we want
+  geom_point()
+```
+
+Here's a brief summary of ggplot2 structure. 
+[ggplot2 structure](diagrams/ggplot_structure.png)
+
+### Adjust our ggplot
+
+Now that we have a base plot that shows our data, we can add layers on to it and adjust it.
+We can adjust the color points using `color`. 
+
+```{r}
+ggplot(
+  tumor_normal_df,
+  aes(
+    x = log_fold_change,
+    y = neg_log_p_val,
+    color = avg_expression
+  ) # We added this argument to color code the points!
+) +
+  geom_point()
+```
+
+Because we have so many points overlapping one another, we will want to adjust 
+the transparency, which we can do with an `alpha` argument. 
+
+```{r}
+ggplot(
+  tumor_normal_df,
+  aes(
+    x = log_fold_change,
+    y = neg_log_p_val,
+    color = avg_expression
+  )
+) +
+  geom_point(alpha = 0.2) # We are using the `alpha` argument to make our points transparent
+```
+
+Notice that we added the alpha within the `geom_point()` function, not to the `aes()`. 
+We did this because we want all of the points to have the same level of transparency, and it will not vary depending on any variable in the data.
+We can also change the background and appearance of the plot as a whole by adding a `theme`.
```

Comment:
Let me know what you think if this. We can add the `geom_hline()` to lower plots as well if you like it. Feel free to change the color or anything else if you have preferences. (dotted line?)
 
```suggestion

We are not limited to a single plotting layer. 
For example, if we want to add a horizontal line to indicate a significance cutoff, we can do that with `geom_hline()`.
For now, we will choose the value of 5.5 (that is close to a Bonferonni correction) and add that to the plot.

```{r}
ggplot(
  tumor_normal_df,
  aes(
    x = log_fold_change,
    y = neg_log_p_val,
    color = avg_expression
  )
) +
  geom_point(alpha = 0.2) + 
  geom_hline(yintercept = 5.5, color = "darkgreen") # we can specify colors by names here
```

We can also change the background and appearance of the plot as a whole by adding a `theme`.
```

Reply:
Added this!